frontend/src/metabase/metadata/pages/DataModel/DataModel.tsx

```diff
@@ -22,28 +27,32 @@ import {
   PreviewSection,
   type PreviewType,
   RouterTablePicker,
-  SegmentsLink,
   SyncOptionsModal,
   TableSection,
 } from "./components";
 import { COLUMN_CONFIG, EMPTY_STATE_MIN_WIDTH } from "./constants";
-import type { RouteParams } from "./types";
+import type { FieldChangeParams, RouteParams } from "./types";
 import { getTableMetadataQuery, parseRouteParams } from "./utils";
 
 interface Props {
-  children?: ReactNode;
-  location: Location;
   params: RouteParams;
 }
 
-export const DataModel = ({ children, location, params }: Props) => {
-  const { databaseId, fieldId, schemaName, tableId } = parseRouteParams(params);
+export const DataModel = ({ params }: Props) => {
+  const {
+    databaseId,
+    fieldId,
+    schemaName,
+    tableId,
+    collectionId,
+    modelId,
+    fieldName,
+  } = parseRouteParams(params);
```

Comment:
Can we please have `ModelDataModel` and `TableDataModel`?
One component supporting both cases is too difficult to reason about.